app/src/main/java/org/thoughtcrime/securesms/preferences/ClearAllDataDialog.kt

```diff
@@ -97,46 +114,50 @@ class ClearAllDataDialog : DialogFragment() {
         }
     }
 
-    private fun clearAllData(deleteNetworkMessages: Boolean) {
+    private fun clearAllData(deletionScope: DeletionScope) {
+
         clearJob = lifecycleScope.launch(Dispatchers.IO) {
-            val previousStep = step
-            withContext(Dispatchers.Main) {
-                step = Steps.DELETING
-            }
+            withContext(Dispatchers.Main) { step = Steps.DELETING }
+
+            if (deletionScope == DeletionScope.DeleteLocalDataOnly) {
 
-            if (!deleteNetworkMessages) {
                 try {
                     ConfigurationMessageUtilities.forceSyncConfigurationNowIfNeeded(requireContext()).get()
                 } catch (e: Exception) {
-                    Log.e("Loki", "Failed to force sync", e)
+                    Log.e(TAG, "Failed to force sync when deleting data", e)
+                    Handler(Looper.getMainLooper()).post {
+                        Toast.makeText(ApplicationContext.getInstance(requireContext()), R.string.errorUnknown, Toast.LENGTH_LONG).show()
+                    }
+                    return@launch
```

Comment:
I think when we have to `return@` we should consider making this a function, as these returns are always hard to follow.

A well named function is self-dcoumenting code.

Reply:
Separated out local delete steps into a separate method.